### PR TITLE
had error creating new entry - troubleshooting

### PR DESCRIPTION
**Pull request by stmcpeters**
hey @derekhouck, i tried to use the create button to create a new blog entry but i'm having trouble. do you mind taking a look?

### CODE REVIEW

File: client/src/components/ListEntries.jsx [hunk 1]
```diff
@@ -8,7 +8,11 @@ const ListEntries = ({ entries, users }) => {
 
   const toggleDetails = (id) => {
     setShowDetails(prevState => prevState === id ? null : id);
-}
+  }
+
+  const onSaveEntry = (newEntry) => {
+    setEntry((entry) => [...entry, newEntry]);
+  }
```
Comment (derekhouck) [inline client/src/components/ListEntries.jsx]: This appears to be the problem. As far as I can tell, `setEntry` does not exist in your code. One level up in `App`, you have `setEntries`. I would recommend moving this function up to that level, adjusting it accordingly, and passing the function down as a prop to `ListEntries`, if that is the correct component for this prop. (see my other comment)

File: client/src/components/ListEntries.jsx [hunk 2]
```diff
@@ -21,6 +25,7 @@ const ListEntries = ({ entries, users }) => {
                 user={users.find((user) => user.username === entry.author_username)}
                 toggleDetails={toggleDetails}
                 showDetails={showDetails}
+                onSaveEntry={onSaveEntry}
```
Comment (derekhouck) [inline client/src/components/ListEntries.jsx]: This does not appear to be used by `Entry`. Did you mean to send this prop to `EntryForm` instead?

File: client/src/components/EntryForm.jsx [hunk 3]
```diff
@@ -52,18 +52,35 @@ const EntryForm = () => {
...
-  // }
+  const handleTagsInputChange = (event) => {
+    const tags = event.target.value;
+    setValuesForm((valuesForm) => ({ ...valuesForm, tags}));
+    if(tags === "") {
+      setError((error) => ({ ...error, errorTags: true}));
+    } else {
+      setError((error) => ({ ...error, errorTags: false}));
+    }
+  }
```
Comment (derekhouck) [inline client/src/components/EntryForm.jsx:63]: Where are you displaying these errors after you set them?

**Review by derekhouck — CHANGES_REQUESTED** [review]
I added some comments on the functions living at the wrong level, props being passed to the wrong component, and a question about displaying errors when they occur. Let me know if you have any questions about any of my feedback.

**Comment by stmcpeters** [comment]
thanks for the feedback! i was able to review my code and implemented all your suggestions. currently, my form submits and displays, and i was also able to show errors to the user when fields are left empty or are written by users not in the database

**Pull request merged**